floop.go

```diff
@@ -49,9 +52,23 @@ func New(conf *Config, input *child.NewInput) (*Floop, error) {
 
 	flp.procInput = input
 	flp.proc, err = child.New(flp.procInput)
+	go flp.listenSignals(flp.proc.Signal)
+
 	return flp, err
 }
 
+func (floop *Floop) listenSignals(sigProcesser func(os.Signal) error) {
+	signalChannel := make(chan os.Signal, 3)
+	signal.Notify(signalChannel, syscall.SIGTERM, syscall.SIGINT)
+	for {
+		sig := <-signalChannel
```

Comment:
No need to put this in a loop since the max number of elements on the channel should be 1. Go straight from signal.Notify to your channel receive. i.e.
```
signal.Notify(signalChannel, syscall.SIGTERM, syscall.SIGINT)
sig := <-signalChannel
```